cumulus/primitives/core/src/lib.rs

```diff
@@ -78,6 +78,43 @@ impl From<MessageSendError> for &'static str {
 	}
 }
 
+/// The origin of an inbound message.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
+pub enum AggregateMessageOrigin {
```

Comment:
Just continuing a thread from the old PR: https://github.com/paritytech/cumulus/pull/2157/files/b53f1b2546558dc607835ac8700f29a04cffc614#r1213170738

I'm trying to think forward into the future about how Snowbridge's usage of MessageQueue can be accommodated in this `AggregateMessageOrigin`, as we will eventually deploy on the BridgeHub parachain hopefully in the near term.

I also see that all the cumulus runtimes are configured to use `xcm_builder::ProcessXcmMessage` as a message processor. However this processor seems to accept messages from all origins, without discriminating whether they are from the HRMP/DMP subsystems, or some other application/pallet installed on the chain. It will mark messages as corrupted if it can't decode them into XCM. So in future this would be problematic for Snowbridge, as we of course have our own message processor.

I have feeling, partially due to the above, but also because of other logistical concerns (which I can expand on), that perhaps Snowbridge should rather have its own instance of the MessageQueue pallet. 

If we go this route, there are two approaches we can take for BridgeHub:
1. Update the `MessageQueue` pallet to support instancing and then install two instances on BridgeHub
2. Snowfork forks MessageQueue and renames the symbols so as not to clash with the Substrate version

Obviously (1) is ideal and I can start working on that if everyone thinks its the right approach.

I realise this comment is starting to go a bit off-topic, so can move the conversation to an issue.

Reply:
I dont quite understand. The format is entirely dependant on the origin. Each origin defines its own queue.  
The message processor can then decide on how to decode the message of each queue. We could add a new `Abridged` variant to the `AggregateMessageOrigin` and then modify the message processor to redirect messages with this origin to a different processor.

By design, it should not be needed to have multiple MQ pallets instantiated.